crates/chain-state/src/in_memory.rs

```diff
@@ -209,25 +241,41 @@ impl BlockState {
         Self(executed_block)
     }
 
-    pub(crate) fn block(&self) -> ExecutedBlock {
+    /// Returns the executed block that determines the state.
+    pub fn block(&self) -> ExecutedBlock {
         self.0.clone()
     }
 
-    pub(crate) fn hash(&self) -> B256 {
+    /// Returns the hash of executed block that determines the state.
+    pub fn hash(&self) -> B256 {
         self.0.block().hash()
     }
 
-    pub(crate) fn number(&self) -> u64 {
+    /// Returns the block number of executed block that determines the state.
+    pub fn number(&self) -> u64 {
         self.0.block().number
     }
 
-    pub(crate) fn state_root(&self) -> B256 {
+    /// Returns the state root after applying the executed block that determines
+    /// the state.
+    pub fn state_root(&self) -> B256 {
         self.0.block().header.state_root
     }
 
-    pub(crate) fn receipts(&self) -> &Receipts {
+    /// Returns the `Receipts` of executed block that determines the state.
+    pub fn receipts(&self) -> &Receipts {
         &self.0.execution_outcome().receipts
     }
+
+    /// Returns a vector of `Receipt` of executed block that determines the state.
+    pub fn flattened_receipts(&self) -> Vec<Receipt> {
```

Comment:
This is required by a method in `BlockReader` is it ok to flatten like this the `Vec<Vec<Option<Receipt>>>` that `Receipts` contains?

Reply:
Unfortunately the `Receipts` type refers to multiple blocks' worth of receipts because the `ExecutionOutcome` is for both single and multiple blocks. In this case we can expect there to be only one element in the outer `Vec`. One thing we could do is `debug_assert` that the list is length one, and `.first().unwrap()` until we find a better way to handle this, does that make sense?

Reply:
makes total sense, done here https://github.com/paradigmxyz/reth/commit/ef07c862f0d3cd625e242a311774b4b8012515f4 ptal

Reply:
makes sense to me, I like the comment